system/jhtree/jhtree.cpp

```diff
@@ -995,14 +956,11 @@ void CKeyIndex::init(KeyHdr &hdr, bool isTLK, bool allowPreload)
         keyHdr->load(hdr);
         offset_t rootPos = keyHdr->getRootFPos();
         Linked<CNodeCache> nodeCache = queryNodeCache();
-        if (allowPreload)
-        {
-            if (nodeCache->getNodeCachePreload() && !nodeCache->isPreloaded(iD, rootPos))
-            {
-                cacheNodes(nodeCache, rootPos, isTLK);
-            }
-        }
-        rootNode = nodeCache->getNode(this, iD, rootPos, NULL, isTLK);
+
+        //The root node is always a branch - but see optimization below
```

Comment:
>but see optimization below

optimization being, pretend root node is a leaf, so in normally bigger leaf cache? Perhaps comment could clarify.

Reply:
reworded the comment